patroni/ha.py

```diff
@@ -758,8 +809,17 @@ def process_healthy_cluster(self):
                 if msg is not None:
                     return msg
 
-                return self.enforce_master_role('no action.  i am the leader with the lock',
-                                                'promoted self to leader because i had the session lock')
+                if self.cluster.is_standby_cluster():
+                    # in case of standby cluster we don't really need to
+                    # enforce anything, since the leader is not a master.
+                    # So just remind the role.
+                    self.enforce_follow_remote_master()
+                    return 'no action. I am the standby leader with the lock'
```

Comment:
It might happen that postgres will be restarted due to change in `recovery.conf`, but it will still report "no action".